frontend/datagathering.cpp

```diff
@@ -157,19 +160,20 @@ void MainWindow::currentInstrumentChanged(const QString &arg1)
 }
 
 // Populate table with cycle data
-void MainWindow::on_cyclesBox_currentTextChanged(const QString &arg1)
+void MainWindow::on_cyclesBox_currentIndexChanged(int index)
 {
     // Handle possible undesired calls
-    ui_->cyclesBox->setDisabled(arg1.isEmpty());
-    ui_->filterBox->setDisabled(arg1.isEmpty());
-    if (arg1.isEmpty())
+    auto currentText = ui_->cyclesBox->itemText(index);
+    ui_->cyclesBox->setDisabled(currentText.isEmpty());
+    ui_->filterBox->setDisabled(currentText.isEmpty());
+    if (currentText.isEmpty())
         return;
 
-    if (arg1[0] == '[')
+    if (currentText[0] == '[')
     {
         for (auto tuple : cachedMassSearch_)
```

Comment:
Could possibly use `std::find_if` and structured bindings here to neaten things up (untested):
```
auto it = std::find_if(cachedMassSearch_.begin(), cachedMassSearch_.end(), [currentText](const auto &&[worker, name]) { return name == currentText.mid(1, currentText.length() - 2); } );
```